serde-generate/runtime/golang/src/lcs/serializer.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Facebook, Inc. and its affiliates
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+package lcs
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/facebookincubator/serde-reflection/serde"
+)
+
+// Serializer implements `serde.Serializer` interface for serializing LCS bytes
+type Serializer struct {
+	buf bytes.Buffer
+}
+
+// NewSerializer creates a new `serde.Serializer`
+func NewSerializer() serde.Serializer {
+	return new(Serializer)
+}
+
+func (s *Serializer) GetBytes() []byte {
+	return s.buf.Bytes()
+}
+
+func (s *Serializer) SerializeLen(value int) error {
+	if value < 0 {
```

Comment:
`value` should be uint64 actually so the check is more `value > MaxLen` => error.

Reply:
Not worth for uint64 because:
1. when input value is int, it's max value == MaxSequenceLength = (1 << 31) - 1, hence no need to validate overflow, although it requires check value < 0 case, which is same effort with checking > MaxLen
2. it tells wrong information to user about how big of the Len could be. Max int is your limit. Where negative is much easier to understand and less likely will do it as len value comparing with pass in more than MaxSequenceLength value, as caller usually just call len(xxx) to get length in golang.
3. in almost 99.99999999999% cases, it is an array or slice, we do len(xxx) to get length, which is int. Using uint64 means an additional type convert to the caller's code every time calls to this function.